main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs

```
@@ -187,9 +187,14 @@ int Run (MonoDevelopOptions options)
 
 			if (!options.NewWindow && startupInfo.HasFiles) {
 				foreach (var file in startupInfo.RequestedFileList) {
-					if (MonoDevelop.Projects.Services.ProjectService.IsWorkspaceItemFile (file.FileName)) {
-						options.NewWindow = true;
-						break;
+					try {
+						if (MonoDevelop.Projects.Services.ProjectService.IsWorkspaceItemFile (file.FileName)) {
+							options.NewWindow = true;
+							break;
+						}
+					} catch (UnauthorizedAccessException ex) {
+						LoggingService.LogError (string.Format ("Unable to check startup file is a workspace item '{0}'", file.FileName), ex);
+						return 1;
```

This will just exit the app?

---

Isn't this the same case where the user has open previous solution set?

---

Nope, that's not the case.

---

Not sure what the best approach is here. Either we give up since we cannot open the file (which is implemented), or we try to continue on, opening the IDE, or passing the sln file to another instance (which changes the behaviour of the code since we would not set NewWindow to true), then the IDE will fail (silently it seems) trying to open that solution.

I tested other areas here:

  - Recent projects - the main IDE window opens but the solution does not load. No error shown to the user.
  - File - Open - Nothing happens on clicking Open. Also seems to cause the dialog to break - unable to close it.

Ideally we would show an error to the user here but right now it seems we silently fail with no information given to the user.